.github/workflows/benckmark.yml

```diff
@@ -2,29 +2,30 @@
 name: ci - benchmark
 
 on:
-  workflow_dispatch:
+  workflow_dispatch: null
   schedule:
     # 3 times day
     # collecting enough data to draw some graphics
-    - cron: '0 1 * * *'
+    - cron: "0 1 * * *"
   # push:
   #   branches:
   #     - master
 jobs:
-  prepare: 
+  prepare:
     name: Prepare build
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
       - uses: actions/setup-node@v2
         with:
           node-version: 14.x
+          cache: npm
```

Comment:
I would like to follow up on this. I assume this is blocked by:
- This PR https://github.com/actions/setup-node/pull/323 as @merceyz suggested
- This question `Is upgrading to pnpm@6.10+ a blocker`? @juanpicado 

Is there any other unknown to solve?

Reply:
We can upgrade, I use the lates pnpm locally should works. Unique blockers are, all should be green and cache should works as it does now. Aside of that I am open to any improvement 👍

Reply:
I wonder why https://github.com/actions/setup-node/pull/323 has not been merged or any feedback. 🤔 ....

One quick solution is fork and apply the patch in our organization, but with the risk the PR is rejected, that's my unique fear.

Reply:
> I wonder why [actions/setup-node#323](https://github.com/actions/setup-node/pull/323) has not been merged or any feedback. 🤔 ....
> 
> One quick solution is fork and apply the patch in our organization, but with the risk the PR is rejected, that's my unique fear.

Since the plan is to use `cache` option to stick to `actions/setup-node`, I don't forking them would be a good solution IMO.

> Trying to understand this, when there is a not a hit in the cache, how does restore-keys work here? Could you give me an example @merceyz ?

Can somebody clarify me how `restore-keys` work in `verdaccio`'s CI?

Reply:
> Can somebody clarify me how restore-keys work in verdaccio's CI?
Maybe this helps
https://docs.github.com/en/actions/advanced-guides/caching-dependencies-to-speed-up-workflows#using-the-cache-action

Reply:
> > Can somebody clarify me how restore-keys work in verdaccio's CI?
> 
> Maybe this helps
> 
> https://docs.github.com/en/actions/advanced-guides/caching-dependencies-to-speed-up-workflows#using-the-cache-action

Thanks for sharing @juanpicado . On top of your link, I would like to share this one, which explains on detail how the cache works: https://docs.github.com/en/actions/advanced-guides/caching-dependencies-to-speed-up-workflows#matching-a-cache-key

Here I guess some questions come:
- How benefitial is `restore-keys` for `verdaccio`?
- Is `verdaccio` getting benefit between jobs of the same workflow? Also between PRs? How many milliseconds?

I think is a good practice to use `restore-keys` as you are doing and... since the execution time is very critical, I'm not sure if we should move this forward... it seems there's more risk than potential benefit. What do you think @juanpicado 

Thanks for sharing all this knowledge with me btw :)

Reply:
I think agree with you. 

There are other things could be improve, maybe use something else instead `upload-artifact` for this step which is reused in more consequent steps.

```
      - name: tar packages
        run: |      
          tar -czvf ${{ github.workspace }}/pkg.tar.gz -C ${{ github.workspace }}/packages .
      - uses: actions/upload-artifact@v2
        with:
          name: verdaccio-artifact
          path: pkg.tar.gz       
```

Reply:
> I think agree with you.

👍🏽  https://github.com/verdaccio/verdaccio/pull/2470#issuecomment-955795121

> There are other things could be improve, maybe use something else instead `upload-artifact` for this step which is reused in more consequent steps.
> 
> ```
>       - name: tar packages
>         run: |      
>           tar -czvf ${{ github.workspace }}/pkg.tar.gz -C ${{ github.workspace }}/packages .
>       - uses: actions/upload-artifact@v2
>         with:
>           name: verdaccio-artifact
>           path: pkg.tar.gz       
> ```
> 
> I mean, not just now in this PR, but as future experiment.

Do you want me to create an issue to talk about this? I would be happy to draft PR something so I get more familiar with verdaccio's workflows. :) @juanpicado